photutils/psf/psfphotometry.py

```diff
@@ -0,0 +1,282 @@
+        """
+        Fit, as appropriate, a compound or single model to a given `groups` of
+        stars. Groups are fitted sequentially from the smallest to the biggest. In
+        each iteration, `image` is subtracted by the previous fitted group. 
+        
+        Parameters
+        ----------
+        image : numpy.ndarray
+            Background-subtracted image.
+        star_groups : list of `~astropy.table.Table`
+            Each `~astropy.table.Table` in this list corresponds to a group of
+            mutually overlapping starts.
+
+        Return
+        ------
+        result_tab : `~astropy.table.Table`
+            Astropy table that contains the results of the photometry.
+        image : numpy.ndarray
+            Residual image.
+        """
+
+        result_tab = Table([[], [], [], [], []],
+                           names=('id', 'group_id', 'x_fit', 'y_fit',
+                                  'flux_fit'),
+                           dtype=('i4', 'i4', 'f8', 'f8', 'f8'))
+
+        star_groups = star_groups.group_by('group_id')
+        
+        groups_order = []
+        for g in star_groups.groups:
+            groups_order.append(len(g))
+
+        N = len(groups_order)
+        while N > 0:
+            curr_order = np.min(groups_order)
+            n = 0
+            N = len(groups_order)
+            while(n < N):
+                if curr_order == len(star_groups.groups[n]):
+                    group_psf = self._get_sum_psf_model(star_groups.groups[n])
+                    x, y, data = self._get_shape_and_data(shape=self.fitshape,\
+                                             star_group=star_groups.groups[n],
+                                                          image=image)
+                    fit_model = self.fitter(group_psf, x, y, data)
+                    print(fit_model)
+                    param_table = self._model_params2table(fit_model,\
+                                                        star_groups.groups[n])
+                    result_tab = vstack([result_tab, param_table])
+                    image = subtract_psf(image, self.psf, param_table)
+                    N = N - 1
+                n += 1
+        return result_tab, image
+
+    def _model_params2table(self, fit_model, star_group):
+        """
+        Place fitted parameters into an astropy table.
+        
+        Parameters
+        ----------
+        fit_model : Fittable2DModel
+        star_group : ~astropy.table.Table
+        
+        Returns
+        -------
+        param_tab : ~astropy.table.Table
+            Table that contains the fitted parameters.
+        """
+
+        param_tab = Table([[], [], [], [], []],
+                          names=('id', 'group_id', 'x_fit', 'y_fit',
+                                 'flux_fit'),
+                          dtype=('i4', 'i4', 'f8', 'f8', 'f8'))
+        if np.size(fit_model) == 1:
+            param_tab.add_row([[star_group['id'][0]],
```

Comment:
It's OK to have `x_0_0` etc for a group that has only one star and then you don't need a special case for it.